sw/device/lib/crypto/impl/rsa/rsa_padding.c

```diff
@@ -403,6 +403,78 @@ static inline void reverse_bytes(size_t input_len, uint32_t *input) {
   HARDENED_CHECK_EQ(i, num_idx);
 }
 
+/**
+ * Copy memory between non-overlapping regions with a randomized byte traversal.
+ *
+ * CAUTION! This function is not considered as secure as `hardened_memcpy` due
+ * to the byte-sized memory accesses vs. 32b word accesses.
+ *
+ * @param dest the region to copy to.
+ * @param src the region to copy from.
+ * @param byte_len, the number of bytes to copy.
+ */
+static status_t randomized_bytecopy(void *restrict dest,
+                                    const void *restrict src, size_t byte_len) {
+  random_order_t order;
+  random_order_init(&order, byte_len);
+
+  size_t count = 0;
+
+  uintptr_t src_addr = (uintptr_t)src;
+  uintptr_t dest_addr = (uintptr_t)dest;
+
+  for (; launderw(count) < byte_len; count = launderw(count) + 1) {
+    size_t byte_idx = launderw(random_order_advance(&order));
+    barrierw(byte_idx);
+
+    uint8_t *src_byte_idx = (uint8_t *)launderw(src_addr + byte_idx);
+    uint8_t *dest_byte_idx = (uint8_t *)launderw(dest_addr + byte_idx);
```

Comment:
```suggestion
    
    // TODO(#8815) byte writes vs. word-wise integrity
    uint8_t *dest_byte_idx = (uint8_t *)launderw(dest_addr + byte_idx);
```

maybe same comment in xor function?

Reply:
Added.